File: unit-tests/new/fortran/test_analyser.f90
! DEPENDS ON: some_file.o

SUBROUTINE external_sub
    RETURN
END SUBROUTINE external_sub

INTEGER FUNCTION external_func()
    external_func = 123
END FUNCTION external_func

MODULE foo_mod
    USE bar_mod, ONLY: foo
    CONTAINS

    SUBROUTINE interrnal_sub
        ! DEPENDS ON: monty_func
        RETURN
    END SUBROUTINE interrnal_sub

    INTEGER FUNCTION internal_func()
        internal_func = 456
    END FUNCTION internal_func

END MODULE foo_mod

File: unit-tests/new/c/test_analyser.py
from pathlib import Path

from fab.dep_tree import AnalysedFile

from fab.tasks.c import CAnalyser
from fab.util import HashedFile


def test_simple_result(tmp_path):
    fpath = Path("test_analyser.c")
    result = CAnalyser().run(HashedFile(fpath, None))

    expected = AnalysedFile(
        fpath=fpath,
        file_hash=None,
        symbol_deps={'monty_func', 'bar_mod'},
        symbol_defs={'external_sub', 'foo_mod', 'external_func'},
        file_deps=set(),
        mo_commented_file_deps={'some_file.c'},
    )
    assert result == expected
